notebooks/analysis/notebooks/Actions.ipynb

```diff
@@ -0,0 +1,156 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "240e130f-4cae-4955-a8ed-a692f5ea9092",
+   "metadata": {},
+   "source": [
+    "## Figuring out what users do when they are searching is made easier by identifying actions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8b240d92-24a9-48a4-a750-f3f111dfa212",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import pandas as pd\n",
+    "\n",
+    "from weco_datascience.reporting import get_recent_data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "34681ded-d95e-4c35-b859-40f1be32e3fa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = get_recent_data(config=os.environ, n=10000, index=\"metrics-conversion-prod\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b48f9fb0-e3d8-4773-a5e8-37662e20be4e",
+   "metadata": {},
+   "source": [
+    "## [Definitions of actions] (https://app.gitbook.com/o/-LumfFcEMKx4gYXKAZTQ/s/MM8zK9GZyFCpy8kvd5Km/analytics-and-reporting/kibana-definitions) are translated into code below ##"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "604ac0df-ea66-4f17-91ac-c30a4a729396",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "concepts = df.loc[(df[\"page.name\"]==\"concepts\")]\n",
+    "concepts[\"action\"]=\"concepts\"\n",
+    "search = df.loc[(df[\"page.name\"]==\"works\") & (df[\"page.query.query\"]!=\" \")]\n",
+    "search[\"action\"]=\"search\"\n",
+    "works =  df.loc[(df[\"page.name\"]==\"work\") & (df[\"page.query.workId\"]!=\" \")]\n",
```

Comment:
There are a couple of spots here where you're checking whether the values in the cells are single spaces. How often is that the case? For example, do we ever actually see `workId=" "` in the data? Are these checks actually achieving what you want them to?

In my experience, assertions like this are usually put in place to exclude null/empty values, but they tend to leak data because they're not _actually_ checking for whether the value is null.  
It might be better to `strip()` all of the string values in those columns, replace any empty strings with `None`, and then check whether each column `isnull()`. That way, you'll also exclude values which are `None`, values which are two spaces, three spaces, tabs, etc etc etc

Reply:
Whoops. On it.